app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/SearchAsyncTask.java

```diff
@@ -22,39 +22,50 @@
 
 import static com.amaze.filemanager.ui.fragments.preference_fragments.PreferencesConstants.PREFERENCE_SHOW_HIDDENFILES;
 
-import java.lang.ref.WeakReference;
 import java.util.regex.Pattern;
 
 import com.amaze.filemanager.file_operations.filesystem.OpenMode;
 import com.amaze.filemanager.filesystem.HybridFile;
 import com.amaze.filemanager.filesystem.HybridFileParcelable;
 import com.amaze.filemanager.ui.fragments.SearchWorkerFragment;
 
-import android.app.Activity;
+import android.content.Context;
 import android.os.AsyncTask;
 import android.util.Log;
 
 import androidx.preference.PreferenceManager;
 
-public class SearchAsyncTask extends AsyncTask<String, HybridFileParcelable, Void>
+public class SearchAsyncTask extends AsyncTask<Void, HybridFileParcelable, Void>
     implements StatefulAsyncTask<SearchWorkerFragment.HelperCallbacks> {
 
   private static final String TAG = "SearchAsyncTask";
 
-  private WeakReference<Activity> activity;
+  /** This necessarily leaks the context */
+  private final Context applicationContext;
+
   private SearchWorkerFragment.HelperCallbacks callbacks;
-  private String input;
-  private OpenMode openMode;
-  private boolean rootMode, isRegexEnabled, isMatchesEnabled;
+  private final String input;
+  private final boolean rootMode;
+  private final boolean isRegexEnabled;
+  private final boolean isMatchesEnabled;
+  private final HybridFile file;
 
   public SearchAsyncTask(
-      Activity a, String input, OpenMode openMode, boolean root, boolean regex, boolean matches) {
-    activity = new WeakReference<>(a);
+      Context context,
+      String input,
+      OpenMode openMode,
+      boolean root,
+      boolean regex,
+      boolean matches,
+      String path) {
+    this.applicationContext = context.getApplicationContext();
     this.input = input;
-    this.openMode = openMode;
     rootMode = root;
     isRegexEnabled = regex;
     isMatchesEnabled = matches;
+
+    this.file = new HybridFile(openMode, path);
+    file.generateMode(this.applicationContext);
```

Comment:
Any reason to generate open mode again, when we already have it while calling.

Reply:
It was the first thing run in the on background. I left it as is.

Reply:
fixed